cdap-gateway/src/main/java/io/cdap/cdap/gateway/router/RouterPathLookup.java

```diff
@@ -158,6 +160,9 @@ private RouteDestination getV3RoutingService(String[] uriParts, AllowedMethod re
       return PREVIEW_HTTP;
     } else if (beginsWith(uriParts, "v3", "system", "serviceproviders")) {
       return APP_FABRIC_HTTP;
+    } else if (beginsWith(uriParts, "v3", "namespaces", null, "credentials") ||
+        beginsWith(uriParts, "v3", "credentials")) {
+      return SYSTEM_WORKER;
```

Comment:
does this work in all modes (ex: standalone)? Even if it does, why are we adding special http handlers in the system worker instead of going through the task framework that other parts of the system use for system workers?

Reply:
as discussed offline, lets separate this out into it's own PR

Reply:
Done